csrc/scheduler/matmul.cpp

```diff
@@ -614,47 +617,96 @@ void scheduleOutputTensor(
       {c},
       {ParallelType::BIDx, ParallelType::BIDy, ParallelType::BIDz});
 }
+
+//! Assign parallelization modes to epilogue inputs. The returned pair of
+//!  collections contains:
+//!   - 1st: TVs that can have all parallelization modes propagated,
+//!   - 2nd: TVs that can have all but vectorization parallel modes
+//!     propagated,
+std::pair<std::vector<TensorView*>, std::vector<TensorView*>>
+partitionEpilogueInputsByParallelizationModes(
+    TensorView* reference_tv,
+    const std::vector<TensorView*>& epilogue_inputs) {
+  const auto& tvs_with_vec = scheduler_utils::getInputsOutputsWithInnerDim(
+      reference_tv, true /*inner_only*/, true /*vectorize_pass*/);
+
+  std::vector<TensorView*> epilogue_input_tvs_with_vec = {};
+  epilogue_input_tvs_with_vec.reserve(tvs_with_vec.size());
+  std::copy_if(
+      tvs_with_vec.cbegin(),
+      tvs_with_vec.cend(),
+      std::back_inserter(epilogue_input_tvs_with_vec),
+      [&epilogue_inputs](TensorView* tv) {
+        return (
+            std::find(epilogue_inputs.cbegin(), epilogue_inputs.cend(), tv) !=
+            epilogue_inputs.cend());
+      });
+
+  // No input tensor can be vectorized so all fusion inputs will be
+  //  parallized without vectorization mode propagated.
+  if (epilogue_input_tvs_with_vec.empty()) {
+    return {epilogue_input_tvs_with_vec, epilogue_inputs};
+  }
+
+  std::vector<TensorView*> epilogue_input_tvs_without_vec = {};
+  for (auto* tv : epilogue_inputs) {
+    if (std::find(
+            epilogue_input_tvs_with_vec.cbegin(),
+            epilogue_input_tvs_with_vec.cend(),
+            tv) == epilogue_input_tvs_with_vec.cend()) {
+      epilogue_input_tvs_without_vec.push_back(tv);
+    }
+  }
```

Comment:
Would the logic be cleaner if we make this like below:
```suggestion
  std::vector<TensorView*> epilogue_input_tvs_with_vec = {};
  std::vector<TensorView*> epilogue_input_tvs_without_vec = {};
  for (auto* tv : epilogue_inputs) {
    if (std::find(
            tvs_with_vec.cbegin(),
            tvs_with_vec.cend(),
            tv) == tvs_with_vec.cend()) {
      epilogue_input_tvs_without_vec.push_back(tv);
    } else {
      epilogue_input_tvs_with_vec.push_back(tv);
    }
  }
```
Then I think that's enough for the entire function.